test/solvers/functional/functional_test.py

```diff
@@ -0,0 +1,482 @@
+
+    # Test gradient of left scalar multiplication
+    assert all_almost_equal((scal * F).gradient(x), scal * (F.gradient(x)),
+                            places=PLACES)
+
+    # Test derivative of left scalar multiplication
+    p = noise_element(space)
+    assert all_almost_equal(((scal * F).derivative(x))(p),
+                            scal * ((F.derivative(x))(p)),
+                            places=PLACES)
+
+    # Test conjugate functional. This requiers positive scaling to work
+    scal = np.random.rand()
+    neg_scal = -np.random.rand()
+
+    with pytest.raises(ValueError):
+        (neg_scal * F).convex_conj
+
+    assert all_almost_equal((scal * F).convex_conj(x),
+                            scal * (F.convex_conj(x / scal)),
+                            places=PLACES)
+
+    # TODO: Add more test for convex conjugate? Old ConvexConjugateArgScaling?
+
+    # Test proximal operator. This requiers scaling to be positive.
+    sigma = 1.0
+    with pytest.raises(ValueError):
+        (neg_scal * F).proximal(sigma)
+
+    sigma = np.random.rand()
+    assert all_almost_equal(((scal * F).proximal(sigma))(x),
+                            (F.proximal(sigma * scal))(x),
+                            places=PLACES)
+
+    # Test left multiplication with zero
+    assert isinstance(0 * F, odl.solvers.ZeroFunctional)
+
+
+def test_right_scalar_multiplication(space):
+    """Test for right and left multiplication of a functional with a scalar."""
+    x = noise_element(space)
+
+    scal = np.random.standard_normal()
+    F = odl.solvers.functional.L2NormSquared(space)
+
+    # Evaluation of right scalar multiplication
+    assert all_almost_equal((F * scal)(x), (F)(scal * x),
+                            places=PLACES)
+
+    # Test gradient of right scalar multiplication
+    assert all_almost_equal((F * scal).gradient(x),
+                            scal * (F.gradient(scal * x)),
+                            places=PLACES)
+
+    # Test derivative of right scalar multiplication
+    p = noise_element(space)
+    assert all_almost_equal(((F * scal).derivative(x))(p),
+                            scal * (F.derivative(scal * x))(p),
+                            places=PLACES)
+
+    # Test conjugate functional: positive, negative and zero scalar
+    scal = np.random.rand()
+    assert all_almost_equal((F * scal).convex_conj(x),
+                            (F.convex_conj(x / scal)),
+                            places=PLACES)
+
+    neg_scal = -np.random.rand()
+    assert all_almost_equal((F * neg_scal).convex_conj(x),
+                            (F.convex_conj(x / neg_scal)),
+                            places=PLACES)
+
+    assert isinstance((F * 0), odl.solvers.ConstantFunctional)
+    assert almost_equal((F * 0)(x), F(space.zero()), places=PLACES)
+
+    # TODO: Add more test for convex conjugate? Old ConvexConjugateFuncScaling?
+
+    # Test proximal operator. This requiers scaling to be positive.
```

Comment:
This is not true... update test and check that it works for 0, + and - (just as the convex conjugate)